src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3283,6 +3291,28 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                         return false;
                     }
 
+                    if (suffix == 'u' || suffix == 'U')
+                    {
+                        if (UInt64.TryParse(strNum, style, NumberFormatInfo.InvariantInfo, out ulong u))
+                        {
+                            ulong testValue = u * (ulong)multiplier;
```

Comment:
From what I can tell `multiplier` is *always* a `long` value in here. All other suffix-specified conversions here do *not* use `biginteger`. That usage only occurs if *no* type specification has occurred and the number is not a `real` value (which I assume means 'contains a decimal point').

So what I've done here essentially reflects how it's already functioning. Should I change this?

Reply:
Try: `18446744073709551615ukb` - that will overflow.

Reply:
Sure, but the same sort of idea with `l` suffix will also overflow currently. This is no different to existing behaviour.

```
PS> 9223372036854775807lkb
At line:1 char:23
+ 9223372036854775807lkb
+                       ~
The numeric constant 9223372036854775807lkb is not valid.
+ CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : BadNumericConstant

PS > 18446744073709551615ukb
At line:1 char:24
+ 18446744073709551615ukb
+                        ~
The numeric constant 18446744073709551615ukb is not valid.
+ CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : BadNumericConstant
```

This correlates precisely with established behaviours. Should I change both these code paths?

Reply:
Sorry, it's been a long time since I wrote that code and didn't look at the surrounding code.  The entire function is guarded with by `checked` and `try/catch` - that code didn't show up in the diff and I forgot about it.

The intent of the parser is to never raise an exception, but apparently there were cases I couldn't reasonably avoid.

So you're fine here.

Reply:
Or I should say - it's not a real problem, but you could still use `BigInteger` here to avoid the exception for in theory better performance.

Reply:
I'll look at that possibility. Is using `bigint` really better than throwing the exception and having it caught?

Reply:
Yes, one additional comparison is better the exception.

Reply:
Sure, but from the little I know `BigInteger` is generally avoided moreso for memory consumption issues; the documentation itself warns that it can cause *memory* overflow issues. I'm not sure that defaulting to it is the wisest cause of action, in all honesty, not least because when parsing any appreciably large value memory consumption may rise noticeably.

Additionally, we would need to define an arbitrary standard value of where to call it a `double` instead and cast back to that if we don't want to store ginormous integers in memory after parsing...